rasa/server.py

```diff
@@ -804,10 +804,18 @@ def validate_request(rjs):
 
         try:
             data = emulator.normalise_request_json(request.json)
-            parse_data = await app.agent.interpreter.parse(
-                data.get("text"), data.get("message_id")
-            )
-            response_data = emulator.normalise_response_json(parse_data)
+            try:
+                parsed_data = await app.agent.parse_message_using_nlu_interpreter(
+                    data.get("text")
+                )
+            except Exception as e:
+                logger.debug("Error while handling message data")
```

Comment:
As you are catching a generic exception, the cause of the error during parsing the message can be anything. I would log the stacktrace, so that users have a chance to debug.
```
logger.debug(traceback.format_exc())
```

Reply:
Hmm, I tested in terminal passing wrong arguments to break the code and the traceback got weird:
```
2019-07-17 10:04:41 DEBUG    rasa.server  - Traceback (most recent call last):
  File "/home/arthur/Documents/rasa/work/rasa/rasa/server.py", line 809, in parse
    {}
  File "/home/arthur/Documents/rasa/work/rasa/rasa/core/agent.py", line 429, in parse_message_using_nlu_interpreter
    return await processor._parse_message(message)
  File "/home/arthur/Documents/rasa/work/rasa/rasa/core/processor.py", line 274, in _parse_message
    if message.text.startswith(INTENT_MESSAGE_PREFIX):
AttributeError: 'dict' object has no attribute 'startswith'

2019-07-17 10:04:41 DEBUG    rasa.server  - Traceback (most recent call last):
  File "/home/arthur/Documents/rasa/work/rasa/rasa/server.py", line 809, in parse
    {}
  File "/home/arthur/Documents/rasa/work/rasa/rasa/core/agent.py", line 429, in parse_message_using_nlu_interpreter
    return await processor._parse_message(message)
  File "/home/arthur/Documents/rasa/work/rasa/rasa/core/processor.py", line 274, in _parse_message
    if message.text.startswith(INTENT_MESSAGE_PREFIX):
AttributeError: 'dict' object has no attribute 'startswith'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/arthur/Documents/rasa/work/rasa/rasa/server.py", line 814, in parse
    400, "ParsingError", "An unexpected error occurred. Error: {}".format(e)
rasa.server.ErrorResponse: (400, 'ParsingError', "An unexpected error occurred. Error: 'dict' object has no attribute 'startswith'")
```

Apparently the second traceback prints the first traceback too, so I decided to just leave the second one. I will update to print both again.